### DISCUSSION TITLE
v0.22.2

### DISCUSSION THREAD

**Discussion by alex-w** [Stellarium/stellarium#2508] [General]
**The installers are located in the "Assets" section at the end of these release notes!**

[![Github Release](https://img.shields.io/github/downloads/Stellarium/stellarium/v0.22.2/total.svg)](https://github.com/Stellarium/stellarium/releases/v0.22.2)

A huge thanks to the people who helped us a lot by reporting bugs!

## Release notes
The Stellarium team is delighted to announce the release of Stellarium 0.22.2.

The major changes of this version:
- Added recognition licenses for skycultures
- Many improvements in the Satellites plugin, esp. display of earth shadow and filtering
- Improvements in AstroCalc: graphs, eclipses, transits
- Improvements in Meteor Showers, Exoplanets and the Navigational Stars plugins
- Updated DSO catalog
- Fixed ANGLE mode for some rare cases
- A large number of bug fixes and closed feature requests and enhancements (see full list of changes).

Behind the scenes, we are preparing for a major upgrade of the internal Qt framework.

Enjoy!

Alexander and Georg

## What's Changed
Full list of changes:
- Added "type" column to WUT list (GH: #2502)
- Added twilights to the nautical navigation data (GH: #2489)
- Added new visual filter for satellites (GH: #1527, #2060)
- Added group of satellites (Gonets)
- Added "Transits of Mercury and Venus across the Sun" computations for AstroCalc tool (GH: #2477)
- Added new filter for communications satellites
- Added feature to recognise the communications data for satellites
- Added GUI for manual addition of comm frequencies for satellites (GH: #2476)
- Added support inclination classifications for geocentric orbits for satellites
- Added feature to adding and restoring the descriptions of satellites
- Added tool to reset TLE sources in Satellites plugin (GH: #2458)
- Added 'altitude' parameter to custom filter of satellites
- Added option to colored invisible satellites: you can toggle behaviour of colorification invisible satellites now
- Added recognition licenses for sky cultures (GH: #2410)
- Added new names to DSO (GH: #2463)
- Added recognition the standard magnitudes for Starlink and OneWeb satellites
- Added ability to select all satellites in the list by one touch
- Added custom filter for satellites (GH: #2457, #2060)
- Added quick selection buttons for DSO (GH: #2425, #2427)
- Added ability to use time mask for naming of screenshots (GH: #1283)
- Added visualisation Earth umbra and penumbra for satellites (GH: #2451, #2360, #1246)
- Added support directional elongation (GH: #2405)
- Added some Barnard outlines (GH: #2412)
- Fixed saving GRS properties
- Fixed missing meteor showers name (GH: #2492)
- Fixed bugs in planetary transits (GH: #2491)
- Fixed orbit validation for observers
- Fixed selecting items from the recent search in Search tool
- Fixed signal/slot issue in Search Tool
- Fixed wrong selection of object in Search Tool/Lists
- Fixed placement for output data on HDPI devices (Observ. plugin)
- Fixed flickering Lunar halo (GH: #2240)
- Fixed selection ANGLE mode: simplify force-switching to ANGLE mode (GH: #2471)
- Fixed missing translations
- Fixed dead links and phrases in sky cultures
- Fixed filtering issue for satellites and update some defaults
- Fixed description for XMM-Newton satellite
- Fixed guessing the groups of satellites
- Fixed slow down of time when the Moon is selected (GH: #2465)
- Fixed eclipse problem (GH: #2464)
- Fixed conversion tool radToHms(): avoiding possible negative zero for value of seconds (GH: #2460)
- Fixed wrong code in radToHMSStr and radToHMSStrAdapt convertors (GH: #2460)
- Fixed work AstroCalc/(Phenomena and RTS) tools with Qt 5.14+ (GH: #2459)
- Fixed Milky Way texture (GH: #2420, #2421)
- Fixed duplicate for NGC 5504
- Fixed stars info: hiding separation info for double stars with "fake values" (GH: #2429)
- Fixed broken signal/slot in Meteor Showers plugin (GH: #2433)
- Fixed 'Nights above horizon' feature in Observability plugin (GH: #2436)
- Fixed rendering orbits of artificial satellites (GH: #2452)
- Fixed crash when some star catalogues deleted or adding (GH: #2404)
- Fixed opacity/magnitude conflict for objects from Barnard catalog (GH: #2408)
- Fixed possible duplicates of DSO names
- Fixed hourly motion of objects (GH: #2505, #2498)
- Fixed show negative radiant drift in RA for meteor showers (GH: #2504)
- Changed setFullScreen() method (GH: #660)
- Changed infostring data for observers: hided the meaningless data
- Changed Satellites plugin: add improvements for visibility prediction of satellites
- Changed behaviour in Satellites plugin: display satellites and their orbits within selected range of altitudes only when related option is activated.
- Changed behaviour in Satellites plugin: Some specific comm terms from the channel description are translatable now
- Changed AstroCalc/Graphs tools: a full refactoring
- Changed AstroCalc/PC tool: refactoring the graphs
- Changed Exoplanets plugin: refactoring the graphs
- Updated filter 'orbit calculation error' in the Satellites plugin
- Updated pulsars catalog
- Updated list of locations
- Updated meteor showers catalog (GH: #2492)
- Updated default list of satellites
- Updated script for InnoSetup (GH: #2450)
- Updated planetary nomenclature
- Updated communications data for satellites
- Updated DSO catalog (v3.16; GH: #2414, #2473)
- Updated list of description of satellites
- Updated data in Meteor Showers catalog (GH: #2467, #2449)
- Updated GUI in Meteor Showers plugin
- Updated default hint color for satellites
- Updated Meteor Showers plugin (GH: #2416)
- Updated DSO names
- Removed QCustomPlot dependence
- Removed scaling mode from shortcuts for Windows (GH: #2450)

## Packages notes
package | note
------------|---------
stellarium-0.22.2.tar.gz | Stellarium 0.22.2 for UNIX/Linux (source code). <br />__MD5:__ 2aafc25f5736b8b5d6d2bcf6d7c2d118 <br />__SHA1:__ 3661b4e8a94e7a520873fa22ddc79f39904eb1c4 <br />__SHA256:__ 31e965d32cafc0fbad212c7ef4efbeac988f909206013554e1fe35123ebb9376
stellarium-0.22.2-win32.exe | Stellarium 0.22.2 for Windows (32-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: 3fd1d44cca2eb993d545c80ac3b258d9 <br />__SHA1__: 1533e590daf7ea2620b9fe281a72cfeb56391a0e <br />__SHA256__: 283f54c87bbf15f80cdfe86ba48be7e6a179b4575fde21133864fa23b679ea87
stellarium-0.22.2-win64.exe | Stellarium 0.22.2 for Windows (64-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: f26cb41c6fac7eb3b9672f17f268a60e <br />__SHA1__: b5601cfbce089a1ef4d1ced85690f68a000a42c7 <br />__SHA256__: 1e14810e7d4eb90e60b842cadb32cb26a9e67ced75aa987bddb2c3bb280b3084
Stellarium-0.22.2-x86_64.zip | Stellarium 0.22.2 for macOS (x86_64; macOS 10.12+) based on Qt 5.12.12 (AppleClang 11.0.0) with media support. <br /> __MD5__: bc024f45ec9a628149b7ce83213085f8 <br />__SHA1__: 690693354180394df1e25b26716e578e01aaa7ca <br />__SHA256__: 5e8c2ee315f8c00a393e7bd22461f9b48355538cec39e1bb771e92a5795bcfb4
Stellarium-0.22.2-arm64.zip | Stellarium 0.22.2 for macOS (arm64; macOS 10.12+) based on Qt 5.15.5 (AppleClang 13.1.6) with media support. <br /> __MD5__: 905d5509c3de3d0255c55fd84a960057 <br />__SHA1__: 1410f9d042bb6379f0e997ee2bc232b4609bd355 <br />__SHA256__: 74acc44f96597d11d92f94015efd19cd18c2ce76e850d90f44c40d636f72ea5f
Stellarium-0.22.2-x86_64.AppImage | Stellarium 0.22.2 for Linux (AMD64; glibc 2.27) based on Qt 5.9.5 (GCC 7.5.0) with media support. <br /> __MD5__: 2a9c61329c85ac91a7c0c5d2cc16aa75 <br />__SHA1__: 7c2880f960c7140d8a808daba4f774a636d3eb5b <br />__SHA256__: 5af698ae3d91f36642e841eae4c8dcf71612899a221f2cc273e170b27cb59730

<hr /><em>This discussion was created from the release <a href='https://github.com/Stellarium/stellarium/releases/tag/v0.22.2'>v0.22.2</a>.</em>

**Comment by tedmickle** [comment]
Stellarium 0.22.1, and the previous two versions cause screen flashes on both PC and Mac platforms - any remedy?

**Reply by gzotti** [reply]
Two or three out of tens of thousands of downloaders have reported some flickering or flashing, without enough technical insight into the actual cause. It is hard to find a reason without being able to reproduce such issues. You could help identifying possible sources, e.g. run without any plugins to check if any plugin is the culprit, full-screen vs. windowed, ..., and file a proper issue report.

**Comment by crisphashbrowns** [comment]
Have you all noticed that between 21.0 and 22.0 that the class "setFlagCardinalsPoints" has been changed "setFlagCardinalPoints"? dropping the pluralization.

Just curious why that was changed! Thanks

**Reply by gzotti** [reply]
Because IMHO the plural form was false here.
